ext/js/language/translator.js

```diff
@@ -514,26 +539,45 @@ export class Translator {
      * @param {import('language').TextProcessorWithId<unknown>[]} textProcessors
      * @param {(import('translation').FindTermsTextReplacement[] | null)[]} textReplacements
      * @param {import('translation-internal').TextCache} textCache
-     * @returns {Set<string>}
+     * @returns {import('translation-internal').VariantAndTextProcessorRuleChainCandidatesMap}
      */
     _getTextVariants(text, textProcessors, textReplacements, textCache) {
-        let variants = new Set([text]);
+        // let variants = new Set([text]);
+
+        /** @type {import('translation-internal').VariantAndTextProcessorRuleChainCandidatesMap} */
+        let variantsMap = new Map([
+            [text, [[]]],
+        ]);
+
         for (const textReplacement of textReplacements) {
             if (textReplacement === null) { continue; }
-            variants.add(this._applyTextReplacements(text, textReplacement));
+            variantsMap.set(this._applyTextReplacements(text, textReplacement), [[]]);
```

Comment:
should we count applying text replacements as text processing?

Reply:
sure, i'll add that